invokeai/frontend/web/src/features/gallery/components/Boards/BoardsList/BoardsList.tsx

```diff
@@ -19,53 +22,106 @@ const overlayScrollbarsStyles: CSSProperties = {
   width: '100%',
 };
 
-type Props = {
-  isOpen: boolean;
-};
-
-const BoardsList = (props: Props) => {
-  const { isOpen } = props;
+const BoardsList = () => {
   const selectedBoardId = useAppSelector((s) => s.gallery.selectedBoardId);
   const boardSearchText = useAppSelector((s) => s.gallery.boardSearchText);
+  const allowPrivateBoards = useAppSelector((s) => s.config.allowPrivateBoards);
   const queryArgs = useAppSelector(selectListBoardsQueryArgs);
   const { data: boards } = useListAllBoardsQuery(queryArgs);
   const filteredBoards = boardSearchText
     ? boards?.filter((board) => board.board_name.toLowerCase().includes(boardSearchText.toLowerCase()))
     : boards;
+  const filteredPrivateBoards = filteredBoards?.filter((board) => board.is_private);
+  const filteredSharedBoards = boards?.filter((board) => !board.is_private);
   const [boardToDelete, setBoardToDelete] = useState<BoardDTO>();
+  const [isPrivateBoardsOpen, setIsPrivateBoardsOpen] = useState(true);
+  const [isSharedBoardsOpen, setIsSharedBoardsOpen] = useState(true);
```

Comment:
Could use `useDisclosure` for these

```suggestion
  const privateBoardsDisclosure = useDisclosure({defaultIsOpen: true})
  const sharedBoardDisclosure = useDisclosure({defaultIsOpen: true})
```

Then in components, `onClick={privateBoardsDisclosure.onToggle}` etc